### ISSUE TITLE
Feature Request: Support encoding/decoding AAC, AAC+ & HE-AAC

### ISSUE THREAD

**Issue by jrlanglois** [WeAreROLI/JUCE#268]
As the overarching AAC codec is extremely popular nowadays, especially in web and video contexts, it would be nice to have official support for it in the framework.

**Comment by hsundt** [comment]
Agreed, would be great to be able to encode m4a via an AudioFormatWriter

**Comment by jrlanglois** [comment]
@hsundt Sure, M4A is a container that uses AAC for its audio streams. But I believe encoding an M4A container would involve writing a video stream as well as the audio stream.

Personally, I would be content with decoding/encoding AAC files (ie: `.aac`), and being able to decode AAC streams from containers. Wouldn't seem very SRP to add encoding for containers to me.
